pkg/networkpolicy/controller.go

```diff
@@ -425,6 +425,15 @@ func (c *Controller) Run(ctx context.Context) error {
 			c.nfq.SetVerdict(*a.PacketID, verdict) //nolint:errcheck
 			return 0
 		}
+		switch packet.proto {
+		case v1.ProtocolUDP:
+		case v1.ProtocolTCP:
+		case v1.ProtocolSCTP:
+		default:
+			// Accept packets that are parsed OK, but not TCP/UDP/SCTP.
+			c.nfq.SetVerdict(*a.PacketID, nfqueue.NfAccept) //nolint:errcheck
+			return 0
```

Comment:
Dan and I were discussed offline, and we concluded that:
1. network policies only care about L4 protocol if you set the Ports fields
2. if not Port field is specified , is legit to assume that the user/admin can consider the network policy as "any" protocol (some from sftim about users wanting to control ICMP is spot on)

As conclusion, I think we set the packet.proto to empty , then srcPort and dstPort will be 0, and IIUIC the network policy logic will be able do the right thing.

This leaves the `parsePacket()` errors to short or malformed packets (module bugs) than we can say we deny by default , as we don't expect at this point to receive a packet from the kernel that is malformed

Reply:
can you just remove this switch case?

Reply:
And just check for an empty v1.Protocol? What is the best if-condition:
```go
  if string(packet.proto) == ""
```
?

Reply:
(I couldn't figure that out, hence the switch)

Reply:
oh, not check at all , just pass the packet as is to the next stage

Reply:
`c.evaluatePacket(packet)` will evaluate the packet only with the IPs information

Reply:
pushed